File: webapp/src/types/Movie.ts
import type {User} from "./User";
import type {Rating} from "./Rating.ts";

export type Movie = {
  id: string;
  title: string;
  description: string;
  createdAt: string;
  updateAt: string;
  owner: User;
  round: number;
  averageRating: number;
  ratings: Rating[]
};

File: backend/src/main/java/com/ridiculousmovies/backend/domain/Movie.java
package com.ridiculousmovies.backend.domain;

import java.time.Instant;
import java.util.ArrayList;
import java.util.List;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class Movie {
  private String id;
  private String title;
  private String description = "";
  private AppUser owner;
  private List<Rating> ratings = new ArrayList<>();
  private Integer round;
  private Instant createdAt;
  private Instant updatedAt;
}
